src/Compilers/Core/Portable/ConstantValue.cs

```diff
@@ -453,6 +457,30 @@ internal static ConstantValueTypeDiscriminator GetDiscriminator(SpecialType st)
             return ConstantValueTypeDiscriminator.Bad;
         }
 
+        public string GetPrimitiveTypeName()
+        {
+            return Discriminator switch
+            {
+                ConstantValueTypeDiscriminator.SByte => "sbyte",
+                ConstantValueTypeDiscriminator.Byte => "byte",
+                ConstantValueTypeDiscriminator.Int16 => "short",
+                ConstantValueTypeDiscriminator.UInt16 => "ushort",
+                ConstantValueTypeDiscriminator.Int32 or ConstantValueTypeDiscriminator.NInt => "int",
```

Comment:
This is not correct for `nint` and `nuint`. The primitive type names are `nint` and `nuint`, respectively.